x-pack/plugins/canvas/public/components/workpad_header/control_settings/kiosk_controls.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  EuiDescriptionList,
+  EuiDescriptionListDescription,
+  EuiDescriptionListTitle,
+  EuiFormLabel,
+  EuiHorizontalRule,
+  EuiLink,
+  EuiSpacer,
+  EuiSwitch,
+  EuiText,
+  EuiFlexGrid,
+  EuiFlexItem,
+} from '@elastic/eui';
+import { timeDurationString } from '../../../lib/time_duration';
+import { CustomInterval } from './custom_interval';
+
+const ListGroup = ({ children }) => <ul style={{ listStyle: 'none', margin: 0 }}>{[children]}</ul>;
+
+export const KioskControls = ({
+  autoplayEnabled,
+  autoplayInterval,
+  onSetEnabled,
+  onSetInterval,
+}) => {
+  const RefreshItem = ({ duration, label }) => (
+    <li>
+      <EuiLink onClick={() => onSetInterval(duration)}>{label}</EuiLink>
+    </li>
+  );
+
+  return (
+    <div>
+      <EuiDescriptionList textStyle="reverse">
+        <EuiDescriptionListTitle>Page cycle settings</EuiDescriptionListTitle>
+        <EuiDescriptionListDescription>
+          <span>Show pages for {timeDurationString(autoplayInterval)}</span>
+        </EuiDescriptionListDescription>
+      </EuiDescriptionList>
+      <EuiHorizontalRule margin="m" />
+
+      <div>
+        <EuiSwitch
+          checked={autoplayEnabled}
+          label="Cycle slides automatically"
+          onChange={ev => onSetEnabled(ev.target.checked)}
+        />
+        <EuiSpacer size="m" />
+      </div>
+
+      <EuiFormLabel>Change cycling interval</EuiFormLabel>
+      <EuiSpacer size="s" />
+
+      <EuiText size="s">
+        <EuiFlexGrid gutterSize="s" columns={2}>
+          <EuiFlexItem>
+            <ListGroup>
+              <RefreshItem duration="5000" label="5 seconds" />
```

Comment:
Suggestion: make this functional, e.g. create a `number[]` constant and write a function that takes a duration and produces the item with the appropriate label.  That will make these `ListGroups` more terse, and allows us to add/remove durations at will.

Reply:
I don't understand how you're suggesting I change this.